inc/form_answer.class.php

```diff
@@ -508,15 +508,8 @@ public function saveAnswers($datas) {
                         ?intval($datas['id'])
                         :-1;
 
-      $query = "SELECT q.`id`, q.`fieldtype`, q.`name`, a.`id` as answer_id
-                FROM glpi_plugin_formcreator_questions q
-                LEFT JOIN glpi_plugin_formcreator_sections s
-                  ON s.`id` = q.`plugin_formcreator_sections_id`
-                LEFT JOIN `glpi_plugin_formcreator_answers` AS a
-                  ON a.`plugin_formcreator_forms_answers_id` = $formanswers_id
-                  AND a.`plugin_formcreator_question_id` = q.`id`
-                WHERE s.`plugin_formcreator_forms_id` = {$datas['formcreator_form']}";
-      $result = $DB->query($query);
+      $question = new PluginFormcreatorQuestion();
+      $questions = $question->getQuestionsFromForm($datas['formcreator_form']);
```

Comment:
get questions of the form with a dedicated method. fields from glpi_plugin_formcreator_answers are retrieved later in the code